.github/workflows/go-releaser.yml

```diff
@@ -12,28 +12,248 @@ permissions:
+      with:
+        version: latest
+        args: release --rm-dist
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GPG_FINGERPRINT: ${{ steps.import_gpg.outputs.fingerprint }}
+        GPG_PASSWORD: ${{ secrets.OCTOPUS_GPG_PASSPHRASE }}
+
+    - name: Calculate version
+      id: calculate-version
+      run: |
+        tag_name=${{ fromJson(steps.goreleaser-release.outputs.metadata).tag }}
+        version=${{ fromJson(steps.goreleaser-release.outputs.metadata).version }}
+        echo "::set-output name=tag_name::$tag_name"
+        echo "::set-output name=version::$version"
+
+    - uses: actions/upload-artifact@v3
+      with:
+        name: artifacts
+        path: |
+          dist/*.zip
+          dist/*.tar.gz
+          dist/*.rpm
+          dist/*.deb
+
+  msi:
+    needs: goreleaser
+    runs-on: windows-latest
+    outputs:
+      msi_file: ${{ steps.buildmsi.outputs.msi }}
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: artifacts
+        path: artifacts/
+
+    - name: Extract octopus.exe
+      id: extract_exe
+      shell: bash
+      run: |
+        cp ./artifacts/*windows_amd64.zip .
+        printf "::set-output name=zip::%s\n" *.zip
+        unzip -o *.zip && rm -v *.zip
+
+    - name: Setup MSBuild
+      id: setupmsbuild
+      uses: microsoft/setup-msbuild@v1.0.3
+
+    - name: Build MSI
+      id: buildmsi
+      shell: bash
+      env:
+        ZIP_FILE: ${{ steps.extract_exe.outputs.zip }}
+        MSBUILD_PATH: ${{ steps.setupmsbuild.outputs.msbuildPath }}
+      run: |
+        name="$(basename "$ZIP_FILE" ".zip")"
+        version="$(echo -e ${{ needs.goreleaser.outputs.version }} | sed 's/-.*$//')"
+        "${MSBUILD_PATH}\MSBuild.exe" ./build/windows/octopus.wixproj -p:SourceDir="$PWD" -p:OutputPath="$PWD" -p:OutputName="$name" -p:ProductVersion="$version"
+
+    - name: Install AzureSignTool
+      run: dotnet tool install --global AzureSignTool
+
+    - name: Sign MSI
+      env:
+        MSI_FILE: ${{ steps.buildmsi.outputs.msi }}
+      shell: powershell
+      run: |
+        $timestampurl = (
+            "http://timestamp.comodoca.com/rfc3161",
+            "http://timestamp.globalsign.com/tsa/r6advanced1", #https://support.globalsign.com/code-signing/code-signing-windows-7-8-and-10,
+            "http://timestamp.digicert.com", #https://knowledge.digicert.com/solution/SO912.html
+            "http://timestamp.apple.com/ts01", #https://gist.github.com/Manouchehri/fd754e402d98430243455713efada710
+            "http://tsa.starfieldtech.com",
+            "http://www.startssl.com/timestamp",
+            "http://timestamp.verisign.com/scripts/timstamp.dll",
+            "http://timestamp.globalsign.com/scripts/timestamp.dll",
+            "https://rfc3161timestamp.globalsign.com/advanced"
+        )
+
+        $ex = $null
+        $signSuccessful = $false
+
+        foreach ($url in $timestampurl) {
```

Comment:
This is going to try and sign the MSI using the first timestamp server in the list (comodo), then if that fails for some reason it'll try globalsign, and so forth.

This means that dependent on the transient availability of timestamp servers, different versions of our MSI might be timestamped by different authorities.

I guess it probably doesn't matter given that it doesn't matter who does the timestamping so long as windows trusts that authority? Seems a touch weird is all

Reply:
I doubt this would cause any issues, and it's also what we do for the current C# CLI

Reply:
ok, all good. Will mark conv as resolved